src/api/java/baritone/api/utils/command/datatypes/RelativeFile.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * This file is part of Baritone.
+ *
+ * Baritone is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Baritone is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Baritone.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package baritone.api.utils.command.datatypes;
+
+import baritone.api.utils.command.helpers.arguments.ArgConsumer;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.InvalidPathException;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import static baritone.api.utils.Helper.HELPER;
+
+public class RelativeFile implements IDatatypePost<File, File> {
+    private final Path path;
+
+    public RelativeFile() {
+        path = null;
+    }
+
+    public RelativeFile(ArgConsumer consumer) {
+        try {
+            path = FileSystems.getDefault().getPath(consumer.getString());
+        } catch (InvalidPathException e) {
+            throw new IllegalArgumentException("invalid path");
+        }
+    }
+
+    @Override
+    public Stream<String> tabComplete(ArgConsumer consumer) {
+        return Stream.empty();
+    }
+
+    /**
+     * Seriously
+     *
+     * @param file File
+     * @return Canonical file of file
+     * @author LoganDark and his hate of checked exceptions
+     */
+    private static File SHUT_THE_FUCK_UP_IOEXCEPTION_NOBODY_LIKES_YOU(File file) {
```

Comment:
This is a family friendly repo. this is unacceptable! also.... https://docs.oracle.com/javase/8/docs/api/java/io/UncheckedIOException.html